config/class-sync.php

```diff
@@ -63,7 +83,65 @@ public function filter_cron_schedules( $schedule ) {
 		return $schedule;
 	}
 
-	public function do_cron() {
+	public function get_blogs_to_sync() {
+		return $this->blogs_to_sync;
+	}
+
+	public function queue_sync_for_blog() {
+		// Queue the sync only from admin or CLI, to keep it out of frontend request path
+		if ( ! is_admin() && ! Context::is_wp_cli() ) {
+			return;
+		}
```

Comment:
@WPprodigy @rinatkhaziev one thing I've also added is to not queue the sync if we're not on admin/WP CLI, like we already do for the `maybe_setup_cron`. 

I didn't include this into the `init_listener` primarily to make it easier to test (I was having troubles defining the test to check that the behavior would actually change if we're not on WP CLI)

Reply:
I like the restriction as it brings about some extra safety.

There's a (small) chance the option is updated via REST/XMLRPC endpoints, which we wouldn't catch here. Maybe that's an okay trade-off though.